docker/Dockerfile.python

```diff
@@ -1,23 +1,45 @@
-FROM fedora:35
+FROM fedora:latest
 
 RUN dnf -y update \
     && dnf -y install \
         cmake \
         gcc-c++ \
         gcc-fortran \
+        gdb \
         git \
         lapack-devel \
         make \
         netcdf-fortran-devel \
-        python-devel \
         pip \
+        python \
+        python-devel \
     && dnf clean all
 
 # Copy the musica code
 COPY . musica
 
-# Install Python dependencies
-RUN pip install -r musica/python/requirements.txt
+# Install Python dependencies from pyproject.toml
+# must do it this way since scikit-build does not have a way to install dependencies from pyproject.toml
+# outside of the build process
+# First, install toml package to parse pyproject.toml
+RUN pip install toml
+
+# Extract dependencies from pyproject.toml
+RUN python -c "import toml; \
+               file = 'musica/pyproject.toml'; \
+               data = toml.load(file); \
+               deps = data.get('build-system', {}).get('requires', []); \
+               [print(dep) for dep in deps]; \
+               " > requirements.txt 
+
+RUN python -c "import toml; \
+               file = 'musica/pyproject.toml'; \
+               data = toml.load(file); \
+               deps = data.get('project', {}).get('optional-dependencies', {}).get('test', []); \
+               [print(dep) for dep in deps]; \
+               " >> requirements.txt
+
+RUN pip install -r requirements.txt
```

Comment:
Is this valid? Looks like `requirements.txt` is deleted and replaced with a call to install `numpy`

Reply:
the `>>` appends and doesn't overwrite